### PR TITLE
introduces FLock for supporting file locking to coordinate work between processes

### PR DESCRIPTION
**Pull request by p0lyn0mial**

### CODE REVIEW

**Comment by p0lyn0mial** [comment]
will be used in https://github.com/openshift/library-go/pull/1130

File: pkg/operator/staticpod/flock.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+// TryLock tries to acquire an exclusive lock on a file
+// until it succeeds, an error is returned or the timeout is reached.
+//
+// The callers MUST call the corresponding Unlock method to release the lock and associated resources except when an error
+// is returned (including the timeout).
+//
+// This method is safe for concurrent access.
+//
+// Note the given timeout shouldn't be less than 1 second.
+func (f *FLock) TryLock(timeout time.Duration) error {
```
Comment (sttts) [inline pkg/operator/staticpod/flock.go:37]: not a context?

Comment (p0lyn0mial) [reply pkg/operator/staticpod/flock.go:37]: we could but then the usage becomes
```
ctx, cancel := context.WithTimeout()
defer cancel()
flock.TryLock(ctx)
```

vs

`flock.TryLock(time.Second * 5)`

**Pull request merged**